web-server/plugins/slycat-parameter-image/js/Components/ControlsDropdown.tsx

```diff
@@ -55,13 +59,16 @@ class ControlsDropdown extends React.Component<ControlsDropdownProps> {
           </h6>
         );
       default:
+        // Get the set_selected function from the item if it's been set, or use the one passed in from props
+        const set_selected_function = item.set_selected ?? this.props.set_selected;
+        const selected = item.selected ?? item.key == this.props.selected;
```

Comment:
Is this to cover if `item.selected` is undefined or null? because unless this is the case, `item.key == this.props.selected` will never get hit

Reply:
Yes! There are many times when item.selected is undefined. In those cases, we determine `selected` by looking at what's set in `props.selected`. But in cases where `item.selected` is already set to `true` or `false`, we just go with that value. This lets you specify the selected item on the item itself or by passing it as a prop, with the item taking precedence. I forget why we needed both methods, but do remember putting that in on purpose.

Reply:
FYI, I added a comment explaining this now.